lib/src/client/common.dart

```diff
@@ -58,10 +58,51 @@ class ResponseFuture<R> extends DelegatingFuture<R>
     return value;
   }
 
-  ResponseFuture(this._call)
-      : super(_call.response
+  ResponseFuture(ClientCall<dynamic, R> call)
+      : _call = call,
+        super(call.response
             .fold(null, _ensureOnlyOneResponse)
             .then(_ensureOneResponse));
+
+  ResponseFuture._clone(Future<R> future, {ClientCall clientCall})
+      : _call = clientCall,
+        super(future);
+
+  ResponseFuture<S> copy<S>(Future<S> future) =>
+      ResponseFuture._clone(future, clientCall: _getClientCall(future, _call));
+
+  /// clientCall maybe be lost when converting from Future to ResponseFuture
+  static ResponseFuture<T> wrap<T>(Future<T> future, {ClientCall clientCall}) {
```

Comment:
Do we actually have any use for `wrap` beyond testing?

Reply:
We need a way to convert `Future` to `ResponseFuture` and vice versa. You can see we use it in the very first step in this example:

```dart
  @override
  ResponseFuture<R> interceptUnary<Q, R>(ClientMethod<Q, R> method, Q request,
      CallOptions options, ClientUnaryInvoker<Q, R> invoker) {
    _invocations.add(InterceptorInvocation(_id, ++_unary, _streaming));
    return ResponseFuture.wrap(
            Future.delayed(Duration(seconds: 1), () => 'dummy'))
        .then((_) => invoker(method, request, _inject(options)))
        .then((foo) async => foo)
        .whenComplete(() => 'complete')
        .then((bar) => bar)
        .catchError((e, s) => print('$e at $s'))
        .timeout(Duration(seconds: 5));
  }```

Reply:
One idea- could you make an extension on Future / ResponseFuture instead?